From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 19 Jan 2018 09:31:01 -0600
Subject: workspaceThumbnail: only update _porthole if the overview is visible

Otherwise it happens that porthole is computed again after that the
overlay is hidden (triggered by a layout reallocation) and thus not
regenerated again afterwards.

https://bugzilla.gnome.org/show_bug.cgi?id=792687

Bug-GNOME: https://bugzilla.gnome.org/show_bug.cgi?id=792687
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/commit/5fcf40b
---
 js/ui/workspaceThumbnail.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/workspaceThumbnail.js b/js/ui/workspaceThumbnail.js
index 7d5d2c0..5565a1e 100644
--- a/js/ui/workspaceThumbnail.js
+++ b/js/ui/workspaceThumbnail.js
@@ -1159,7 +1159,7 @@ var ThumbnailsBox = new Lang.Class({
     // The "porthole" is the portion of the screen that we show in the
     // workspaces
     _ensurePorthole() {
-        if (!Main.layoutManager.primaryMonitor)
+        if (!Main.layoutManager.primaryMonitor || !Main.overview.visible)
             return false;
 
         if (!this._porthole)
